openapi2conv/openapi2_conv.go

```diff
@@ -848,27 +895,30 @@ func FromV3Operation(swagger *openapi3.Swagger, operation *openapi3.Operation) (
 		result.Parameters = append(result.Parameters, r)
 	}
 	if v := operation.RequestBody; v != nil {
-		parameters := FromV3RequestBodyFormData(operation.RequestBody)
-		if len(parameters) > 0 {
-			result.Parameters = append(result.Parameters, parameters...)
-		} else {
-			// Find parameter name that we can use for the body
-			name := findNameForRequestBody(operation)
-			if name == "" {
-				return nil, errors.New("could not find a name for request body")
-			}
-			r, err := FromV3RequestBody(swagger, name, v)
-			if err != nil {
-				return nil, err
+		// Find parameter name that we can use for the body
+		name := findNameForRequestBody(operation)
+		if name == "" {
+			return nil, errors.New("could not find a name for request body")
+		}
+
+		bodyOrRefParameters, formDataParameters, consumes, err := FromV3RequestBodies(name, v, &swagger.Components)
+		if err != nil {
+			return nil, err
+		}
+		if len(*formDataParameters) != 0 {
+			result.Parameters = append(result.Parameters, *formDataParameters...)
+		} else if len(*bodyOrRefParameters) != 0 {
```

Comment:
Again, I feel there should be an error when there are both formDataParameters and bodyOrRefParameters. Thoughts?

Maybe a system a registerable callbacks (so the user can tune behavior in this case and when picking which body to use)  can be added in later, when someone requires it.

Reply:
Sure I'll change it. I don't need it for my own use case

Reply:
I mean lets do best effort, as discussed a bit above. If one needs more then we can always plug in callbacks later.